en/components/grid.md

```diff
@@ -75,39 +73,127 @@ Let's turn the **autoGenerate** property off and define the columns collection i
+
+`igxHeader` targets the column header providing as a context the column object itself.
+
+```html
+...
+<igx-column field="Name">
+    <ng-template igxHeader let-column>
+        {{ column.field | uppercase }}
+    </ng-template>
+</igx-column>
+...
+```
+
+`igxCell` applies the provided template to all cells in the column. The context object provided in the template consists of the cell value provided implicitly and the cell object itself.
+
+```html
+...
+<igx-column field="Name">
+    <ng-template igxCell let-value>
+        {{ value | titlecase }}
+    </ng-template>
+</igx-column>
+...
+```
+
+In the snippet above we "take" a reference to the implicitly provided cell value. This is sufficient if you just want to present some data and maybe apply some custom styling or pipe transforms over the value of the cell. However even more useful is to take the `IgxGridCellComponent` object itself as shown below:
+
+```html
+<igx-grid #grid [data]="data">
+    <igx-column dataType="string" field="Name">
+        <ng-template igxCell let-cell="cell">
+            <!-- Implement row deleting inside the cell template itself -->
+            <span tabindex="0" (keydown.delete)="grid.deleteRow(cell.rowIndex)">{{ cell.value | titlecase }}</span>
+        </ng-template>
+    </igx-column>
+    <igx-column dataType="boolean" field="Subscribtion">
+        <ng-template igxCell let-cell="cell">
+            <!-- Bind the cell value through the ngModel directive and update the data source when the value is changed in the template -->
+            <input type="checkbox" [ngModel]="cell.value" (ngModelChange)="cell.update($event)" />
+        </ng-template>
+    </igx-column>
+<igx-grid>
+```
+
+The column also accepts one last template that will be used when a cell is in edit mode. As with the other column templates, the provided context object is again the cell value and the cell object itself. Of course in order to have your template show up make sure to set the `editable` property of the column to true.
+
+```html
+<igx-column dataType="number" editable="true" field="Price">
+    <ng-template igxCellEditor let-cell="cell">
+        <label for="price">
+            Enter the new price tag
+        </label>
+        <input name="price" type="number" [ngModel]="cell.value" (ngModelChange)="cell.update(convertToNumber($event))" />
+    </ng-template>
+</igx-column>
+```
+
+Make sure to check the API for the [IgxGridCellComponent](#igxgridcellcomponent) in order to get accustomed with the provided properties you can use in your templates.
+
+Each of the column templates can be changed programmatically at any point through the column object itself. For example in the code below, we have declared two templates for our user data. In our Typescript code we'll get references to the templates themselves and then based on some condition we will render the appropriate template for the column in our application.
```

Comment:
"... through the `IgxColumnComponent` object itself."

Also refer to Typescript as TypeScript.